### ISSUE TITLE
{{link-to tagName="button"}} Linting

### ISSUE THREAD

**Issue by hergaiety** [ember-template-lint/ember-template-lint#1231]
My preface for this is that ideally we wouldn't see any `{{link-to` that has a `tagName`, and we are moving towards a world of Angle bracket notation. However, as it currently stands we still support curly bracket notation and I still see real world examples of `{{link-to`'s with custom `tagName`'s.

Most notably, the following template code will miss the https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/require-button-type.md require-button-type rule.

``` html.hbs
{{link-to 'new.path' tagName="button"}}foo{/link-to}}
```

**Comment by lifeart** [comment]
This is absolutely valid code:
```hbs
<LinkTo @route="new.path" @tagName="button">foo</LinkTo>
```

**Comment by MelSumner** [comment]
> This is absolutely valid code:
> 
> ```
> <LinkTo @route="new.path" @tagName="button">foo</LinkTo>
> ```

what do you mean by valid? It's certainly an a11y error. 

If it's an `<a>` element, it should go to a URL. If it's a `button` then it should be some sort of toggle (open a modal, a menu, switch a tab, save, etc.).

**Comment by lifeart** [comment]
Sorry for confusing, I mean it "syntactically" valid. @MelSumner I seen use cases when modal is independent route like `users/edit/ -> users/edit/confirm-delete` and such.

**Comment by lifeart** [comment]
Looks like we can't simply bind "type" html attribute to `{{link-to}}`, we need to write code like
```hbs
{{#link-to "application" tagName="button" type="button" attributeBindings=(array "type")}} {Link-To} {{/link-to}}
```
